File: 8kyu 3/replaceDots.js
const replaceDots = function (str) {
   return str.split(".").join("-")
}


var replaceDots = function (str) {
   return str.replace(/\./g, '-');
}


var replaceDots = function (str) {
   let answer = "";
   for (let i = 0; i < str.length; i++) {
      answer += (str[i] !== ".") ? str[i] : "-";
   }
   return answer;
}


var replaceDots = function (str) {
   return str.replace(/[.]/g, '-');
}


console.log(replaceDots("one.two.three"), "one-two-three", "Sorry, try again :-(");

File: Python/8py 3/replace_dots.py
"""
|--------------------------------------------------------------------------
| FIXMEE: Replace all dots
|--------------------------------------------------------------------------
|
"""

def replace_dots(s):
   return s.replace('.', '-')


replace_dots = lambda str: str.replace('.', '-')



import re
def replace_dots(str):
   return re.sub(r"\.", "-", str)


def replace_dots(str):
   return'-'.join(str.split('.'))


import re
def replace_dots(s):
   return re.sub("[.]", "-", s)


print(replace_dots(""), "")
print(replace_dots("no dots"), "no dots")
print(replace_dots("one.two.three"), "one-two-three")
print(replace_dots("........"), "--------")